.github/workflows/check-commit-message.yml

```diff
@@ -0,0 +1,38 @@
+name: check-commit-message
+
+run-name: Check commit messages by ${{ github.actor }} on ${{ github.ref }}
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    branches:
+      - master
+
+jobs: 
+  # push events have commits as a list of actions
+  check_commit_on_push:
+    if: github.event_name == 'push'
+    name: check commit message
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check commit message python script
+        run: python .github/check_commit_message.py
+        shell: bash
+        env:
+          COMMITS: ${{ toJSON(github.event.commits) }}
+
+  complain_if_not_squash:
```

Comment:
Could be a bit aggressive for users...

Reply:
Yup, finally found it - I looked before but didn't understand that it would force squash merge if I disabled every other type of merge since it was called "Allow"